Source/WebInspectorUI/UserInterface/Workers/HeapSnapshot/HeapSnapshot.js

```diff
@@ -298,6 +307,8 @@ HeapSnapshot = class HeapSnapshot
         console.assert("node" in shortestPath[shortestPath.length - 1], "Path should end with a node");
 
         return shortestPath.map((component) => {
+            if (component.node && component.rootReason)
+                return Object.assign(this.serializeNode(component.node), { rootReason: component.rootReason });
             if (component.node)
                 return this.serializeNode(component.node);
```

Comment:
i think a cleaner way of doing this that doesnt involve `Object.assign` would be
```js
if (component.node) {
    let node = this.serializeNode(component.node);
    if (component.rootReason)
        node.rootReason = component.rootReason;
    return node;
}
```